dubbo-registry/dubbo-registry-multiple/src/main/java/org/apache/dubbo/registry/multiple/MultipleRegistry.java

```diff
@@ -200,9 +195,27 @@ public void unsubscribe(URL url, NotifyListener listener) {
                 urls.addAll(tmpUrls);
             }
         }
+        urls = removeDuplicate(urls);
         return urls;
     }
 
+    protected static List<URL> removeDuplicate(List<URL> urls) {
+        Set<List<String>> set = new HashSet<>();
+        List<URL> dedupList = new ArrayList<>();
+        for (URL url : urls) {
+            List<String> tmp = new ArrayList<>();
+            tmp.add(url.getAddress());
+            tmp.add(url.getProtocol());
+            tmp.add(url.getParameter("group"));
+            tmp.add(url.getParameter("version"));
+            if (!set.contains(tmp)) {
+                set.add(tmp);
+                dedupList.add(url);
+            }
```

Comment:
It's not the best way to check if two URLs are equal, you can refer [URL#equals](https://github.com/apache/dubbo/blob/3.0/dubbo-common/src/main/java/org/apache/dubbo/common/URL.java#L1432)

Reply:
I rebased the branch from 3.0 to Master.And maybe what you want to say is that these parameters are not enough to complete the deduplication correctly?

Reply:
And I found that If I put the deduplication code in MultipleNotifyListenerWrapper.notify(), the code cannot pass the unit test in MultipleRegistry2S2RTest:212, because the parameters of NotifyListener.notify() are deduplicated,  not the same as the parameters declared by the test.

Reply:
This is a multi-instance problem, and the community is discussing a reasonable solution to solve this issue. There are some thing duscussed in community, such as

+ How to define an unique instance
+ What resources need to be shared in a multi-instance scenario
+ How to define the lifecycle of the instance

If you are interested in this problem, welcome to join the metting!

+ Metting tool: [DingDing](https://www.dingtalk.com/)
+ Meeting Code: `864 912 87261`
+ Reference: [Agenda](https://www.yuque.com/apache-dubbo/dubbo3/wz70lz)